src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,69 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel;
+import com.revrobotics.RelativeEncoder;
+import com.spikes2212.command.drivetrains.TankDrivetrain;
+import com.spikes2212.command.drivetrains.smartmotorcontrollerdrivetrain.SparkMaxTankDrivetrain;
+import com.spikes2212.control.PIDSettings;
+import com.spikes2212.dashboard.ChildNamespace;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import frc.robot.RobotMap;
+
+import java.util.function.Supplier;
+
+public class Drivetrain extends SparkMaxTankDrivetrain {
```

Comment:
add distance per pulse constant
you need to add constants for gear ratio, wheel diameter (in inches), and inches to centimeters 

after this use `encoder.setConversionFactor` in constructor